File: repository-remote/src/main/java/com/andyanika/translator/repository/remote/YandexApi.java
package com.andyanika.translator.repository.remote;

import io.reactivex.Observable;
import retrofit2.http.POST;
import retrofit2.http.Query;

interface YandexApi {
    @POST("translate")
    Observable<YandexTranslationResponse> translate(@Query("key") String key,
                                                    @Query("text") String text,
                                                    @Query("lang") String direction);
}

File: repository-remote/src/main/java/com/andyanika/translator/repository/remote/YandexApi.kt
package com.andyanika.translator.repository.remote

import com.andyanika.translator.common.models.TranslateResult
import retrofit2.http.POST
import retrofit2.http.Query

internal interface YandexApi {
    @POST("translate")
    suspend fun translate(
        @Query("key") key: String?,
        @Query("text") text: String?,
        @Query("lang") direction: String?
    ): YandexTranslationResponse
}
